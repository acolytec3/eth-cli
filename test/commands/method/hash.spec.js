const MethodHashCommand = require('../../../src/commands/method/hash')

describe('hash', () => {
  let stdoutResult

  beforeEach(() => {
    stdoutResult = []
    jest
      .spyOn(process.stdout, 'write')
      .mockImplementation(val => stdoutResult.push(require('strip-ansi')(val.toString('utf8'))))
  })

  afterEach(() => jest.restoreAllMocks())

  it(`Should run 'hash' with empty args and flags and throw an error.`, async () => {
    await expect(MethodHashCommand.run()).rejects.toThrow()
  })

  it(`Should run 'hash' and throw an error.`, async () => {
    await expect(MethodHashCommand.run([])).rejects.toThrow(
      'Missing 1 required arg:\n' +
        'signature  The given signature.\n' +
        'See more help with --help'
    )
  })

  it(`Should run 'hash --help' and throw an error.`, async () => {
    await expect(MethodHashCommand.run(['--help'])).rejects.toThrow('EEXIT: 0')
  })

  it(`Should run 'hash -- 'updateAppInstance'' and get success`, async () => {
    await MethodHashCommand.run(['updateAppInstance'])
    expect(stdoutResult).toMatchSnapshot()
  })
})
