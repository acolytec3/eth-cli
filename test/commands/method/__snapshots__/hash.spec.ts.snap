// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hash Should run 'hash -- 'updateAppInstance'' and get success 1`] = `
Array [
  "2ab3e7c9
",
]
`;
